modules/nwtc-library/tests/test_NWTC_IO_FileInfo.F90

```diff
@@ -73,12 +74,46 @@ subroutine test_initfileinfo2()
         "line 3",  &
         "line 4"   &
     /)
+    ! make the last character a + so a trim does not reduce this string length
+    tmpstring=input_strings(5)
+    tmpstring(MaxFileInfoLineLen+1:MaxFileInfoLineLen+1) = 'a'
+    input_strings(5)=tmpstring
     call InitFileInfo( input_strings, file_info_type, error_status, error_message )
-
     @assertEqual(num_lines, file_info_type%NumLines)
     @assertEqual(num_files, file_info_type%NumFiles)
     @assertEqual( 4, error_status )
     
 end subroutine
 
-end module
\ No newline at end of file
+@test
+subroutine test_initfileinfo3()
+    USE ISO_C_BINDING, only: C_CHAR, C_NULL_CHAR
+
+    ! This case should result in zero error status.
+    ! It attempts to initialize FileInfoType with a C_NULL_CHAR delimited string 
+
+    integer, parameter :: num_lines = 7
+    integer, parameter :: num_files = 1
+
+    integer(IntKi) :: error_status
+    character(ErrMsgLen) :: error_message
+    character(kind=C_CHAR,len=MaxFileInfoLineLen*2) :: input_string
+    type(FileInfoType) :: file_info_type
+    integer :: i
+
+    ! Note: the rest of the input string is empty.  That should pass ok
+    input_string =   "line  0"//C_NULL_CHAR//   &
+                     "line  1"//C_NULL_CHAR//   &
+                     "line  2"//C_NULL_CHAR//   &
+                     "line  3"//C_NULL_CHAR//   &
+                     "line  4"//C_NULL_CHAR//   &
+                     "line  5"//C_NULL_CHAR//   &
+                     "line  6"//C_NULL_CHAR
+    call InitFileInfo( input_string, file_info_type, error_status, error_message )
+    @assertEqual(num_lines, file_info_type%NumLines)
+    @assertEqual(num_files, file_info_type%NumFiles)
+    @assertEqual( 0, error_status )
```

Comment:
It would be good to add a test on the contents of the resulting array after its parsed. You'll probably have to make another array of input strings since `input_string` here is all one string.